### ISSUE TITLE
New Rule: Do not use virtual for field events

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-dotnet#2589]
### Description

I came across a weird behavior while fixing some legacy code that could be caught easily with an analyzer. Overriding a field event results in unexpected behaviors (as mentioned in [MS doc](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-raise-base-class-events-in-derived-classes))

> Do not declare virtual events in a base class and override them in a derived class. The C# compiler does not handle these correctly and it is unpredictable whether a subscriber to the derived event will actually be subscribing to the base class event.

This scenario is quite uncommon so the implementation might not be top priority for you but IMO the severity is quite high.

The idea would be to raise an issue on the override part when overridden with a field-like event.

```csharp
public class Base
{
    public virtual event EventHandler MyEvent;

    public void Foo()
    {
        MyEvent?.Invoke(this, EventArgs.Empty);
    }
}

public class Sub : Base
{
    public override event EventHandler MyEvent;
}

public class Sub2 : Base
{
	public override event EventHander MyEvent
	{
		add
		{
			// do some more logic
			base.MyEvent += value;
		}
		remove
		{
			// do some more logic
			base.MyEvent -= value;
		}
	}
}

public class Program
{
	public void Main()
	{
		Base sub1 = new Sub();
		sub1.MyEvent += (s, e) => { /* not reached */ };
		sub1.Foo();

		Base sub2 = new Sub();
		sub2.MyEvent += (s, e) => { /* reached */ };
		sub2.Foo();
	}
}
```

Cheers,
Amaury

**Comment by andrei-epure-sonarsource** [comment]
thanks, @Evangelink , it's great to have you back as a contributor 😄
cc @nicolas-harraudeau-sonarsource

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1986](https://sonarsource.atlassian.net/browse/NET-1986)

[NET-1986]: https://sonarsource.atlassian.net/browse/NET-1986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
